### ISSUE TITLE
[Camera] CameraException(cameraAccess, CAMERA_ERROR (3): submitRequestList:283: Camera 0: Got error Invalid argument (-22) after trying to submit capture request)CameraException(cameraAccess, CAMERA_ERROR (3): submitRequestList:283: Camera 0: Got error Invalid argument (-22) after trying to submit capture request)

### ISSUE THREAD

**Issue by carman247** [flutter/flutter#88775]
Getting this error when taking picture with both front and rear camera on Huawei Mate 10 Lite.

I was hoping that Camera 0.9.0 would have this fix included.

I've had to change this (`android/src/main/java/io/flutter/plugins/camera/Camera.java`)
```
      captureSession.stopRepeating();
      captureSession.abortCaptures();
      Log.i(TAG, "sending capture request");
```
to this
```
      if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.N) {
        captureSession.stopRepeating();
        // captureSession.abortCaptures();
      }
```

### Flutter doctor
```
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.2.1 20D75 darwin-x64, locale en-GB)
    • Flutter version 2.2.3 at /Users/chris/Developer/flutter
    • Framework revision f4abaa0735 (8 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/chris/Library/Android/sdk
    • Platform android-30, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.59.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0
```

### Device info
```
Huawei Mate 10 Lite
• Model RNE-L21
• Build number RNE-L21 8.0.0.320(C432)
• EMUI version 8.0.0
• Android version 8.0.0
```

**Comment by TahaTesser** [comment]
Hi @carman247 

> Getting this error when taking picture with both front and rear camera on Huawei Mate 10 Lite.

Is the issue only with this device? Can you please provide your a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) and `flutter run --verbose` logs running the minimal code sample? 
Thank you

**Comment by carman247** [comment]
> Hi @carman247
> 
> > Getting this error when taking picture with both front and rear camera on Huawei Mate 10 Lite.
> 
> Is the issue only with this device? Can you please provide your a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) and `flutter run --verbose` logs running the minimal code sample?
> Thank you

No, it's not just that device .. I've had reports for other Huawei devices as well, including: 
• Huawei MediaPad T3 10 (Android 7)
• MediaPad T3 10 (Android 7)

Seems to be a problem with devices using Android 7 and 8 but potentially others as well ... I'm not sure if my change has solved the problem for those devices on Android 7 but I will be monitoring error reports.

There are potentially other devices and versions of Android it affects.

As for a minimal reproducible code sample, the error is present in the example app provided with the current camera package.

**Comment by TahaTesser** [comment]
Hi @carman247 
I just attempted to reproduce on emulator running Android 7, no issue

**Comment by carman247** [comment]
> Hi @carman247
> I just attempted to reproduce on emulator running Android 7, no issue

Is there any chance that it might not affect emulators and potentially affects real devices, like the ones I've listed?

**Comment by TahaTesser** [comment]
@carman247 
All devices you have listed are Huawei running Android 7, we can update if somebody confirms the issue with other Android 7 physical devices. Unfortunately, I don't have Android 7/8 devices, no errors on Android 11.

**Comment by carman247** [comment]
> @carman247
> All devices you have listed are Huawei running Android 7/8, we can update if somebody confirms the issue with other Android 7 physical devices. Unfortunately, I don't have Android 7/8 devices, no errors on Android 11.

Ok, thanks. I've managed to fix the problem for myself, and if anyone else is having the same issue hopefully they'll see this.

**Comment by henri-henri** [comment]
I have this issue too, HUAWEI NOVA 2 LITE LDN-LX2 with Android 8

**Comment by bhanuka96** [comment]
same issue on Huawei P8 Lite 2015 model device. I am using `0.9.3` version.

**Comment by bhanuka96** [comment]
Please fix this issue asap

**Comment by darekjarek** [comment]
Have the same issue on Huawei Mate10 Lite (android 8). Please fix.

**Comment by szuch** [comment]
Same issue happening on Huawei Y6 2018, Android 8.0.0

**Comment by doanbh** [comment]
Have the same issue on Huawei Nova 2i

**Comment by venzeo** [comment]
Patch code suggested by @carman247 works. 
Tested on Huawei Y7 running Android 8, and NO issues found on devices Pixel 2, Pixel 4, Moto G5 and few more.

Can somebody ( @TahaTesser  ) add this patch into main version of the camera plugin?

**Comment by stuartmorgan** [comment]
> Can somebody ( @TahaTesser ) add this patch into main version of the camera plugin?

Anyone can submit a PR to fix this; the plugin is open source.

@carman247 Is there a reason you're vending a fork of the plugin but not submitting a PR (with an explanation of why this change works)?

**Comment by carman247** [comment]
> > Can somebody ( @TahaTesser ) add this patch into main version of the camera plugin?
> 
> Anyone can submit a PR to fix this; the plugin is open source.
> 
> @carman247 Is there a reason you're vending a fork of the plugin but not submitting a PR (with an explanation of why this change works)?

Hi, sorry for the late reply. I've been away. Seems like this is still an issue with the latest (0.9.4+5) version and so I will submit a PR.

**Comment by ocean-c** [comment]
I also have this problem. Huawei Mate 8, Android 8.0.0 with Camera 0.9.4+5

**Comment by mauriziopinotti** [comment]
While we wait for an official patch, I've provided a forked plugin with the fix here: https://github.com/easyhour/plugins

You can add this to your `pubspec.yaml` to replace the official camera plugin:

```
dependency_overrides:
  # Waiting for https://github.com/flutter/flutter/issues/88775
  # Applied https://github.com/flutter/plugins/pull/4572
  camera:
    git:
      url: https://github.com/easyhour/plugins.git
      path: packages/camera/camera
      ref: huawei_camera_fix
```

It's tested and working on Huawei P9 (EVA-AL10) with Android 8.

**Comment by carman247** [comment]
@mauriziopinotti I've already submitted a pull request to fix this problem (https://github.com/flutter/plugins/pull/4572).

What changes have you made in your forked repo? They seem different to what I've implemented so would be interesting to test yours out.

**Comment by mauriziopinotti** [comment]
> @mauriziopinotti I've already submitted a pull request to fix this problem ([flutter/plugins#4572](https://github.com/flutter/plugins/pull/4572)).
> 
> What changes have you made in your forked repo? They seem different to what I've implemented so would be interesting to test yours out.



> @mauriziopinotti I've already submitted a pull request to fix this problem ([flutter/plugins#4572](https://github.com/flutter/plugins/pull/4572)).
> 
> What changes have you made in your forked repo? They seem different to what I've implemented so would be interesting to test yours out.

Nothing significant, it's just your patch adapted to latest codebase: done that for convenience because your PR seems blocked.

**Comment by fnoceda** [comment]
> 

Same problem with Samsung Android 7, this not solve for my :-(

**Comment by carman247** [comment]
@fnoceda Hi, can you please confirm whether this fixes the problem or not?

```
  camera:
    git:
      url: https://github.com/carman247/plugins
      path: packages/camera/camera
      ref: camera-fix-camera-access-exception
```

**Comment by fnoceda** [comment]
> Please report back here if it does not solve it [flutter/plugins#4572](https://github.com/flutter/plugins/pull/4572)

Hi carman, yes this work for me, you save my life, thanks a lot for you help! Do you will maintain up this branch? or i need download for use?

**Comment by carman247** [comment]
@fnoceda I need to finish some bits on the PR ... You can use the git ref for now but when the PR is complete the change will be included in the camera package

**Comment by fnoceda** [comment]
> @fnoceda I need to finish some bits on the PR ... You can use the git ref for now but when the PR is complete the change will be included in the camera package

Hi again @carman247 when I take pictures in a Motorola G5 Android 8, the pictures are blurred this is my initialized configuration, so i can't improve more the resolution.

  _cameraController = CameraController(
        description,
        ResolutionPreset.max,
        enableAudio: false,
        imageFormatGroup:
            (isIOS) ? ImageFormatGroup.bgra8888 : ImageFormatGroup.yuv420,
      );

any suggestion?
